### PR TITLE
Point browser to SCA for signAgreement

### PR DESCRIPTION
**Pull request by sparkiegeek**
Note this depends on [a SCA MP](https://code.launchpad.net/~adam-collard/software-center-agent/add-cors-headers/+merge/322856) before it can be deployed. I figure it's worth getting :eyes: on this first :man_shrugging: 

Instead of talking to the backend proxy for `signAgreement` POST directly to SCA. This removes one case of the security issue highlighted in #272

### CODE REVIEW

File: src/common/actions/register-name.js [hunk 1]
```diff
@@ -65,16 +65,15 @@ export async function getPackageUploadRequestMacaroon() {
 }
 
 async function signAgreement(root, discharge) {
-  const response = await fetch(`${BASE_URL}/api/store/agreement`, {
+  const response = await fetch(`${conf.get('STORE_API_URL')}/agreement`, {
```
Comment (bartaz) [inline src/common/actions/register-name.js]: We have `STORE_API_URL` as const already.

File: src/server/routes/store.js [hunk 2]
```diff
@@ -1,13 +0,0 @@
...
-import {
-  signAgreement
-} from '../handlers/store';
-
-const router = Router();
-
-router.use('/store/agreement', json());
-router.post('/store/agreement', signAgreement);
-
-export default router;
```
Comment (bartaz) [inline src/server/routes/store.js]: If router is now fully removed shouldn't `handlers/store.js` be removed as well?

Comment (sparkiegeek) [inline src/server/routes/store.js]: Yes! Killed, thanks.

File: src/common/actions/register-name.js (hunk 1 above)
Comment (sparkiegeek) [inline src/common/actions/register-name.js]: Oops, re-used.

**Pull request merged**